autoload/airline/extensions/branch.vim

```diff
@@ -174,17 +174,23 @@ function! airline#extensions#branch#head()
   let l:hg_head = s:get_hg_branch()
 
   let l:file = expand("%:p")
+  " Do not get untracked flag if we are modifying a directory.
+  let l:is_file_and_not_dir = (findfile(l:file) == l:file)
```

Comment:
Perhaps instead of an possible expensive call to findfile, isdirectory would be better?